src/main/java/Parser.java

```diff
@@ -0,0 +1,93 @@
+import exception.EmptyDescriptionException;
+import exception.EmptyTimeException;
+import exception.UnknownCommandException;
+import exception.UnknownTaskException;
+import exception.UnknownTimeException;
+
+/**
+ * Represents the Parser class, that interprets user input commands.
+ */
+public class Parser {
+
+    /**
+     * Interprets user input text, into understandable commands.
+     *
+     * @param inputText Commands input by user.
+     * @return Command understood by Chatbot.
+     */
+    protected Command commandHandler(String inputText) {
+        try {
+            String[] splitArr = inputText.split(" ");
+            String command = splitArr[0];
+
+            if (command.equals("find")) {
+                String keyword = splitArr[1];
+                return new Command("find", keyword);
+            } else if (command.equals("bye")) {
+                return new Command("bye");
+            } else if (command.equals("list")) {
+                return new Command("list");
+            } else if (command.equals("done")) {
+                if (splitArr.length == 1) {
+                    throw new UnknownTaskException("No task number entered");
+                }
+                int taskNo = Integer.parseInt(splitArr[1]) - 1;
+                return new Command("done", taskNo);
+            } else if (command.equals("todo")) {
+                if (inputText.length() <= 5) {
+                    throw new EmptyDescriptionException("No Description entered");
+                }
+                String description = inputText.substring(5);
+                return new Command("todo", new Todo(description));
+
+            } else if (command.equals("deadline")) {
+                if (inputText.length() <= 9) {
+                    throw new EmptyDescriptionException("No Description entered");
+                }
+                if (splitArr.length == 1) {
+                    throw new UnknownTimeException("No by time added");
+                }
+                String[] newSplitArr = inputText.substring(9).split("/by ");
+                if (newSplitArr.length == 1) {
+                    throw new EmptyTimeException("No time entered");
+                }
+                String description = newSplitArr[0];
+                String by = newSplitArr[1];
+                return new Command("deadline", new Deadline(description, by));
+            } else if (command.equals("event")) {
+                if (inputText.length() <= 6) {
```

Comment:
Just a suggestion: instead of checking for length, use substring to split the code, thereafter, check if empty.